src/components/Plugin/PluginProposalsPage.tsx

```diff
@@ -240,35 +240,93 @@ class SchemeProposalsPage extends React.Component<IProps, null> {
-  wrappedComponent: SchemeProposalsPage,
+const SubscribedPluginProposalsPage = withSubscription<IProps, SubscriptionData>({
+  wrappedComponent: PluginProposalsPage,
   loadingComponent: <Loading/>,
   errorComponent: null,
 
   checkForUpdate: (oldProps, newProps) => {
-    return oldProps.scheme.id !== newProps.scheme.id;
+    return oldProps.pluginState.id !== newProps.pluginState.id;
   },
 
   createObservable: async (props: IExternalProps) => {
     const arc = getArc();
-    const dao = props.daoState.dao;
-    const schemeId = props.scheme.id;
+    const dao = new DAO(arc, props.daoState.id);
+    const pluginId = props.pluginState.id;
 
     // this query will fetch al data we need before rendering the page, so we avoid hitting the server
     let bigProposalQuery;
     if (props.currentAccountAddress) {
       bigProposalQuery = gql`
-        query ProposalDataForSchemeProposalsPage {
+        query ProposalDataForPluginProposalsPage {
           proposals (where: {
-            scheme: "${schemeId}"
+            scheme: "${pluginId}"
             stage_in: [
               "${IProposalStage[IProposalStage.Boosted]}",
               "${IProposalStage[IProposalStage.PreBoosted]}",
               "${IProposalStage[IProposalStage.Queued]}"
               "${IProposalStage[IProposalStage.QuietEndingPeriod]}",
             ]
           }){
-            ...ProposalFields
+            id
+            accountsWithUnclaimedRewards
+            boostedAt
+            closingAt
+            confidenceThreshold
+            createdAt
+            dao {
+              id
+              schemes {
+                id
+                address
+              }
+            }
+            description
+            descriptionHash
+            executedAt
+            executionState
+            expiresInQueueAt
+            genesisProtocolParams {
+              id
+              activationTime
+              boostedVotePeriodLimit
+              daoBountyConst
+              limitExponentValue
+              minimumDaoBounty
+              preBoostedVotePeriodLimit
+              proposingRepReward
+              queuedVotePeriodLimit
+              queuedVoteRequiredPercentage
+              quietEndingPeriod
+              thresholdConst
+              votersReputationLossRatio
+            }
+            scheme {
+              ...PluginFields
+            }
+            gpQueue {
+              id
+              threshold
+              votingMachine
+            }
+            organizationId
+            preBoostedAt
+            proposer
+            quietEndingPeriodBeganAt
+            stage
+            stakesFor
+            stakesAgainst
+            tags {
+              id
+            }
+            totalRepWhenCreated
+            totalRepWhenExecuted
+            title
+            url
+            votesAgainst
+            votesFor
+            votingMachine
+            winningOutcome
```

Comment:
ouch!  Can we document in the code why all this is needed?

Seems very fragile!

Reply:
```
// NOTE: We cannot use the fragment to reduce the boilerplate here because
// we're using nested where filters for votes, stakes, and gpRewards. These fields are already
// present in the fragment. See here for a solution: https://github.com/daostack/arc.js/issues/471
```

Reply:
Added this comment